app/src/main/java/com/mstudio/superheroarch/Model/MainViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.mstudio.superheroarch
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class MainViewModel : ViewModel() {
+
+    private val apiRick: ApiRick = ApiService.retrofit.create(ApiRick::class.java)
+
+    private val _characters = MutableLiveData<List<Character>>()
```

Comment:
Try to use MutableStateFlow instead of LiveData. LiveData is linked to android, MutableStateFlow is similar but kotlin dependent, better.

Reply:
Will replace: LiveData with MutableStateflow, in the next commit